roles/confluent.zookeeper/templates/zookeeper.properties.j2

```diff
@@ -2,6 +2,22 @@
 {% for key, value in zookeeper.properties.items() %}
 {{key}}={{value}}
 {% endfor %}
+
+{% if zookeeper_sasl_digestmd5_enabled %}
+quorum.auth.enableSasl=true
+quorum.auth.learnerRequireSasl=true
+quorum.auth.serverRequireSasl=true
+quorum.auth.learner.loginContext=QuorumLearner
+quorum.auth.server.loginContext=QuorumServer
+quorum.cnxn.threads.size=20
+requireClientAuthScheme=sasl
 {% for host in groups['zookeeper'] %}
-server.{{ hostvars[host]['zookeeper_id'] | default(groups.zookeeper.index(host) + 1)}}={{ host }}:{{zookeeper_peer_port}}:{{zookeeper_leader_port}}
+authProvider.{{ hostvars[host]['zookeeper_id'] | default(groups.zookeeper.index(host) + 1)}}=org.apache.zookeeper.server.auth.SASLAuthenticationProvider
+{% endfor %}
+{% endif %}
+
+#Usually hostname or fqdn is defined in /etc/hosts as 127.0.1.1, it makes zookeeper listen on it and loosing ability to accept incoming connection
```

Comment:
Comments in jinja are like this:
```
{# Usually hostname or fqdn is defined in /etc/hosts as 127.0.1.1, it makes zookeeper listen on it and loosing ability to accept incoming connection #}
```

But to understand what you are saying here, if we keep it as inventory_hostname then that host can't accept connections from other zookeeper nodes? How has this been functioning this far?

Reply:
If host has it's own name in /etc/hosts with 127.0.0.1 address (which is common), zookeeper starts listening on 127.0.0.1 address. Issue you linked has more detailed description.
To be honest, it is not related to zookeeper auth too.

Reply:
> If host has it's own name in /etc/hosts with 127.0.0.1 address (which is common), zookeeper starts listening on 127.0.0.1 address. Issue you linked has more detailed description.
> To be honest, it is not related to zookeeper auth too.

Hi, can you confirm the scenario and OS version you are running where by you have the hostname set to 127.0.0.1?  I ask because generally this is set to `localhost` as it's a loopback adapter that you don't want to resolve to the actual hostname.  Generally when I've seen it set to something else, it's due to some type of DNS forwarding server setup.  I know in the past that Debian did some interesting things around this due to bugs etc, but I haven't seen this behaviour on other distributions.

Reply:
It is debian 10 and it has such /etc/hosts by default:

```
127.0.0.1       localhost
127.0.1.1       servershostname

# The following lines are desirable for IPv6 capable hosts
::1     localhost ip6-localhost ip6-loopback
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters

```

Reply:
Thanks for the info on this, I've definitely learned something here.  I did some digging and discovered that Arch linux does the same thing, but according to the MAN pages, it appears to be optional.  RHEL for example does not do this.  Either way, looks like a good change to make, in order to be more flexible across different distros.  Thanks again for the example.